batchflow/named_expr.py

```diff
@@ -568,3 +568,61 @@ def get(self, batch=None, pipeline=None, model=None, parallel=False):   # pylint
         if parallel:
             return self.name.get(batch=batch, pipeline=pipeline, model=model)
         return self
+
+class I(NamedExpression):
+    """ Iteration number
+
+    Parameters
+    ----------
+    name : str
+        'c', 'current' - current iteration number, default.
+        'm', 'max', maximum' - total number of iterations to be performed.
+                               If total number is not defined, raises an error.
+        'r', 'ratio' - current iteration divided by a total number of iterations.
+
+    Raises
+    ------
+    ValueError
+        * If `name` is not valid.
+        * If `name` is 'm' or 'r' and total number of iterations is not defined.
+    Examples
+    --------
+    ::
+
+        I()
+        I('m')
+        P(R('normal', loc=0, scale=I('ratio')*100)
+    """
+    def __init__(self, name='c'):
+        super().__init__(name, mode=None)
+        self._allowed_names = {
```

Comment:
You don't need allowed names. Just check for the first chars

Reply:
I can indeed, but I don't like that it would allow for misleading things like `I(my_current_iter)`

Reply:
No, I meant you could compare the passed name (my_current_iter or max) against the beginning of predefined names (maximum, current). And if they fit (max is a substring of maximum, while my_current_iter is not a substring of anything), you use the matching predefined name, otherwise fail.